crypto/asn1/a_mbstr.c

```diff
@@ -272,12 +270,15 @@ static int out_utf8(unsigned long value, void *arg)
 static int type_str(unsigned long value, void *arg)
 {
     unsigned long types;
+    const int native = ossl_fromascii(value);
```

Comment:
I have no objection to having `ossl_fromascii` accept a `long` argument which would resolve the implicit conversion problem here.  A `long long` would be okay, although they'll be a code size and performance hit on many platforms by doing so.

The `ossl_isXXX` functions have the same problem.  They are passed unsigned long arguments in several places.

Any suggestions as to a fix?